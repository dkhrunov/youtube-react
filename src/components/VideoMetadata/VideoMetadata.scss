@import '../../styles/shared.scss';

.video-metadata {
	color: #707070;
	font-size: 16px;

	h3 {
		font-weight: 400;
		color: $text-color-dark;
	}

	.video-stats {
		display: flex;
		justify-content: space-between;
		align-items: center;
		width: 100%;

		.video-actions {
			display: flex;
			align-items: center;

			& > *:not(:last-child) {
				margin-right: 8px;
			}
		}
	}
}