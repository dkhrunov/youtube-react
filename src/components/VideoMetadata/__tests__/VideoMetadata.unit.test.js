import React from 'react';
import { shallow } from 'enzyme';
import VideoMetadata from '../VideoMetadata';

describe('Renders VideoMetadata component', () => {
	test('renders without props', () => {
		const wrapper = shallow(<VideoMetadata />);
		expect(wrapper).toMatchSnapshot();
	});

	test('renders with view count', () => {
		const wrapper = shallow(<VideoMetadata viewCount={100234} />);
		expect(wrapper).toMatchSnapshot();
	});
});
