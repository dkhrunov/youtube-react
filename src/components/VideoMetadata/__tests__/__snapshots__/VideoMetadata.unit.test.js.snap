// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders VideoMetadata component renders with view count 1`] = `
<div
  className="video-metadata"
>
  <h3>
    Video title
  </h3>
  <div
    className="video-stats"
  >
    <span>
      100,234
       views
    </span>
    <div
      className="video-actions"
    >
      <Rating
        dislikeCount={100}
        likeCount={1000}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        lavelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="add circle"
        />
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;

exports[`Renders VideoMetadata component renders without props 1`] = `
<div
  className="video-metadata"
>
  <h3>
    Video title
  </h3>
  <div
    className="video-stats"
  >
    <span>
      0
       views
    </span>
    <div
      className="video-actions"
    >
      <Rating
        dislikeCount={100}
        likeCount={1000}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        lavelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="add circle"
        />
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;
